pkg/utils/share/share.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package share
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/liqotech/liqo/pkg/utils/getters"
+	liqolabels "github.com/liqotech/liqo/pkg/utils/labels"
+	"github.com/liqotech/liqo/pkg/utils/slice"
+)
+
+// Resources contains the resources quantities.
+type Resources map[string]resource.Quantity
+
+// UnavailableMsg is the message to show when a resource is not available.
+const UnavailableMsg = "Unavailable"
+
+// WellKnownResources contains the well known resources.
+var WellKnownResources = []string{
+	"cpu",
+	"memory",
+	"ephemeral-storage",
+	"pods",
+}
+
+// CPU returns the CPU quantity as a string.
+func (r Resources) CPU() string {
+	if cpu, ok := r["cpu"]; ok {
+		result := cpu.ScaledValue(resource.Milli)
+		return fmt.Sprintf("%dm", result)
+	}
+	return UnavailableMsg
+}
+
+// Memory returns the memory quantity as a string.
+func (r Resources) Memory() string {
+	if mem, ok := r["memory"]; ok {
+		result := float64(mem.ScaledValue(resource.Mega)) / 1000
+		return fmt.Sprintf("%.3fGB", result)
+	}
+	return UnavailableMsg
+}
+
+// Pods returns the pods quantity as a string.
+func (r Resources) Pods() string {
+	if pods, ok := r["pods"]; ok {
+		return pods.String()
+	}
+	return UnavailableMsg
+}
+
+// Storage returns the storage quantity as a string.
+func (r Resources) Storage() string {
+	if storage, ok := r["ephemeral-storage"]; ok {
+		result := float64(storage.ScaledValue(resource.Mega)) / 1000
+		return fmt.Sprintf("%.2fGB", result)
+	}
+	return UnavailableMsg
+}
+
+// Others returns the resources that are not well known.
+func (r Resources) Others() map[string]string {
+	result := map[string]string{}
+	for k, v := range r {
+		if !slice.ContainsString(WellKnownResources, k) {
+			k = cases.Title(language.Und).String(k)
```

Comment:
I would avoid converting to title case, as the result is strange for more complex resources: `Devices.kubevirt.io/Vhost-Net`